isis/src/mro/apps/himos/himos.cpp

```diff
@@ -0,0 +1,317 @@
+#include "himos.h"
+
+#include "Camera.h"
+#include "Cube.h"
+#include "FileList.h"
+#include "IException.h"
+#include "Pvl.h"
+#include "IString.h"
+#include "Longitude.h"
+#include "OriginalLabel.h"
+#include "Process.h"
+#include "ProgramLauncher.h"
+#include "TProjection.h"
+
+#include <vector>
+
+using namespace std;
+
+namespace Isis {
+  //functions in the code
+  void CompareLabels(Pvl &match, Pvl &comp);
+
+
+  void himos(UserInterface &ui) {
+
+    // Get the list of cubes to mosaic
+    FileList flist(ui.GetFileName("FROMLIST"));
+
+
+    vector<Cube *> clist;
+    try {
+      if(flist.size() < 1) {
+        QString msg = "the list file [" + ui.GetFileName("FROMLIST") +
+                      "does not contain any data";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
+
+      // open all the cube and place in vector clist
+
+      for(int i = 0; i < flist.size(); i++) {
+        Cube *c = new Cube();
+        clist.push_back(c);
+        c->open(flist[i].toString());
+      }
+
+
+
```

Comment:
stupid nitpick, but large chunks of whitespace is kind of an eyesore. I imagine it's just part of the original .cpp.